lib/countly.js

```diff
@@ -2956,9 +2958,11 @@
                 }
                 // fallback on error
                 xhr.onreadystatechange = function() {
-                    if (this.readyState === 4 && this.status >= 200 && this.status < 300) {
-                        if (typeof callback === 'function') {
-                            callback(false, params, this.responseText);
+                    //check if operations done, status code is acceptable and the response is JSON 
+                    if (this.readyState === 4 && this.status >= 200 && this.status < 300 && this.responseType === "json") {
```

Comment:
"responseType" should not be checked as that is not reliable and does not guarantee anything

Reply:
Neither does the lack of the correcty resposnse type mean that the response is not a json.